gerrit_python_tools/gerrit.py

```diff
@@ -624,13 +682,30 @@ def send_upstream(self, downstream, upstream):
         approvals = self.get_approvals(downstream.SSH())
 
         # Check to see if comment has necessary approvals.
-        if not self.is_upstream_approved(approvals):
-            msg = ("Could not send to upstream: One or more labels"
-                   " not approved.")
+        if self.is_forced():
+            if not self.is_release_approved(approvals):
+                msg = "Could not send to upstream: Release not approved."
+                logger.debug("Change %s: %s" % (self.change_id, msg))
+                ssh.exec_once('gerrit review -m %s %s'
+                              % (pipes.quote(msg), self.revision))
+                return
+            msg = ("***WARNING***\n%s (%s) has requested a forced upstream "
+                   "push. Bypassing all votes except for Release...\n\n"
```

Comment:
It is accusatory. I don't want people to use this.


----
Andrew Kerr
QA, OpenStack
Data Fabric Group

NetApp
919.476.4899 Direct Phone
Andrew.Kerr@netapp.com<mailto:Andrew.Kerr@netapp.com>

On Nov 18, 2015, at 11:25 AM, Goutham Pacha Ravi <notifications@github.com<mailto:notifications@github.com>> wrote:


In gerrit_python_tools/gerrit.py<https://github.com/NetApp-openstack-dev/gerrit-python-tools/pull/3#discussion_r45221747>:

> @@ -624,13 +682,30 @@ def send_upstream(self, downstream, upstream):
>          approvals = self.get_approvals(downstream.SSH())
>
>          # Check to see if comment has necessary approvals.
> -        if not self.is<http://self.is>_upstream_approved(approvals):
> -            msg = ("Could not send to upstream: One or more labels"
> -                   " not approved.")
> +        if self.is<http://self.is>_forced():
> +            if not self.is<http://self.is>_release_approved(approvals):
> +                msg = "Could not send to upstream: Release not approved."
> +                logger.debug("Change %s: %s" % (self.change_id, msg))
> +                ssh.exec_once('gerrit review -m %s %s'
> +                              % (pipes.quote(msg), self.revision))
> +                return
> +            msg = ("***WARNING***\n%s (%s) has requested a forced upstream "
> +                   "push. Bypassing all votes except for Release...\n\n"


Why is this a WARNING, after its been done? IMHO I think it sounds accusatory and should be dropped.

-
Reply to this email directly or view it on GitHub<https://github.com/NetApp-openstack-dev/gerrit-python-tools/pull/3/files#r45221747>.